sdk/program/src/message.rs

```diff
@@ -335,7 +335,7 @@ impl Message {
         let mut writable_keys = vec![];
         let mut readonly_keys = vec![];
         for (i, key) in self.account_keys.iter().enumerate() {
-            if self.is_writable(i) {
+            if self.is_writable(i) && !crate::sysvar::is_sysvar_id(key) {
```

Comment:
Wouldn't it make sense to reject all together the ones which write-lock sysvar earlier?
e.g. in sanitize or like the ones with duplicate keys here:
https://github.com/solana-labs/solana/blob/34bebb7d0/runtime/src/accounts.rs#L794-L796

Reply:
That's an option, it would be a change of behavior which would break some applications on mainnet.